Sources/MlemMiddleware/API Client/ApiClient.swift

```diff
@@ -198,21 +206,28 @@ extension ApiClient: ActorIdentifiable {
 extension ApiClient {
     /// Cache for ApiClient--exception case because there's no ApiType and it may need to perform ApiClient bootstrapping
     class ApiClientCache: CoreCache<ApiClient> {
+        func getCacheId(for baseUrl: URL, with token: String?) -> Int {
+            var hasher: Hasher = .init()
+            hasher.combine(baseUrl)
+            hasher.combine(token)
+            return hasher.finalize()
+        }
         func createOrRetrieveApiClient(for baseUrl: URL, with token: String?) -> ApiClient {
-            let cacheId: Int = {
-                var hasher: Hasher = .init()
-                hasher.combine(baseUrl)
-                hasher.combine(token)
-                return hasher.finalize()
-            }()
-            
-            if let client = retrieveModel(cacheId: cacheId) {
+            if let client = retrieveModel(cacheId: getCacheId(for: baseUrl, with: token)) {
                 return client
             }
             
             let ret: ApiClient = .init(baseUrl: baseUrl, token: token)
             cachedItems[ret.cacheId] = .init(content: ret)
             return ret
         }
+        
+        // Should ONLY be used when we get a new token for THE SAME account
+        func changeToken(for baseUrl: URL, oldToken: String?, newToken: String?) {
+            let oldCacheId = getCacheId(for: baseUrl, with: oldToken)
+            let newCacheId = getCacheId(for: baseUrl, with: oldToken)
```

Comment:
Both generate with `oldToken`--should it be

```suggestion
            let oldCacheId = getCacheId(for: baseUrl, with: oldToken)
            let newCacheId = getCacheId(for: baseUrl, with: newToken)
```
?

Reply:
Yep, nice catch!